### PR TITLE
Initial player color changing code

### PR DESCRIPTION
**Pull request by Flapperkewiet**
This is the inital version for changing the player color.

Currently the the player chooses a random color when joining a game.

Any feedback is welcome.

### CODE REVIEW

File: NebulaClient/GameLogic/PlayerManager.cs
Comment (hubastard) [inline NebulaClient/GameLogic/PlayerManager.cs:11]: @hubastard Note to self, I'll need to add a `editorconfig` file with the code formatting rules of my project to prevent issues like different tabs identation that we see here.

File: NebulaClient/GameLogic/PlayerManager.cs
Comment (hubastard) [inline NebulaClient/GameLogic/PlayerManager.cs:26]: Please add // TODO: in from of this comment so we can keep track of this

File: NebulaModel/Packets/Players/PlayerColorChanged.cs
Comment (hubastard) [inline NebulaModel/Packets/Players/PlayerColorChanged.cs:8]: We should probably use a Float3 for the color since we don't need to send the Alpha channel. It will always be 1.

File: NebulaServer/Player.cs
Comment (hubastard) [inline NebulaServer/Player.cs:15]: Store it as Float3 also

File: NebulaServer/Server.cs
Comment (hubastard) [inline NebulaServer/Server.cs:108]: @hubastard Note to self, should add support of the Log class in the server code also to have the same API everywhere.

File: NebulaServer/Server.cs
Comment (hubastard) [inline NebulaServer/Server.cs:109]: This can be `ReliableUnordered` since its not that important.

File: NebulaServer/GameLogic/PlayerManager.cs
Comment (hubastard) [inline NebulaServer/GameLogic/PlayerManager.cs:62]: Both send packet of color update should be in the mode `ReliableUnordered`

**Review by hubastard — CHANGES_REQUESTED** [review]

File: NebulaServer/GameLogic/PlayerManager.cs
Comment (Flapperkewiet) [reply NebulaServer/GameLogic/PlayerManager.cs:62]: Wouldn't that make it possible for the PlayerColorChanged packet to arrive before the RemotePlayerJoined packet?

File: NebulaClient/GameLogic/PlayerManager.cs
Comment (Flapperkewiet) [reply NebulaClient/GameLogic/PlayerManager.cs:11]: So, i change the tabs to spaces right?

File: NebulaServer/GameLogic/PlayerManager.cs
Comment (Mushroom) [reply NebulaServer/GameLogic/PlayerManager.cs:62]: Perhaps it would be better to wait until my changes are merged in, as I fix this issue with a handshake packet.

File: NebulaClient/GameLogic/PlayerManager.cs
Comment (hubastard) [reply NebulaClient/GameLogic/PlayerManager.cs]: Yeah, but if you merge master in your branch you should be able to now click the 'clean' button in a file to reformat it using the new .editorconfig file.

File: NebulaServer/GameLogic/PlayerManager.cs
Comment (hubastard) [reply NebulaServer/GameLogic/PlayerManager.cs:62]: I've merged all the other PRs into master if you want to update work code with it.

File: NebulaClient/GameLogic/PlayerManager.cs
Comment (Flapperkewiet) [reply NebulaClient/GameLogic/PlayerManager.cs]: Do you mean right-clicking solution ==> Clean Solution?

**Comment by Flapperkewiet** [comment]
I believe everything should be fixed now? Otherwise, let me know.

**Pull request merged**
